networkx/algorithms/dag.py

```diff
@@ -287,3 +289,89 @@ def is_aperiodic(G):
         return g==1
     else:
         return g==1 and nx.is_aperiodic(G.subgraph(set(G)-set(levels)))
+
+def transitive_closure(G):
+    """ Returns transitive closure of a directed graph
+
+    The transitive closure of G = (V,E) is a graph G+ = (V,E+) such that
+    for all v,w in V there is an edge (v,w) in E+ if and only if there
+    is a non-null path from v to w in G.
+
+    Parameters
+    ----------
+    G : NetworkX DiGraph
+      Graph
+
+    Returns
+    -------
+    TC : NetworkX DiGraph
+       Graph
+
+    Raises
+    ------
+    NetworkXError
+      If G is not directed
+
```

Comment:
It can also raise `NetworkXUnfeasible` if the input DiGraph contains a cycle. We should explicitly test this.